src/main/java/org/candlepin/subscriptions/resource/HostsResource.java

```diff
@@ -148,7 +178,21 @@ public HostReport getHosts(
       validateBeginningAndEndingDates(start, end);
 
       String month = InstanceMonthlyTotalKey.formatMonthId(start);
-
+      // We depend on a "reference UOM" in order to filter out instances that were not active in
+      // the selected month. This is also used for sorting purposes (same join). See
+      // org.candlepin.subscriptions.db.HostSpecification#toPredicate and
+      // org.candlepin.subscriptions.db.HostRepository#findAllBy.
+      Measurement.Uom referenceUom;
+      if (SORT_TO_UOM_MAP.containsKey(sort)) {
+        referenceUom = SORT_TO_UOM_MAP.get(sort);
+      } else {
+        referenceUom =
+            Optional.ofNullable(tagProfile.uomsForTag(productId.toString()))
+                .orElse(List.of())
+                .stream()
+                .findFirst()
+                .orElse(null);
```

Comment:
@mstead it's the "reference UOM" which is *always* used for the filter (find instances having metrics for the specified month), and in the case that no sort is specified, it's not used in the sort.

I could move the defaulting to the `HostSpecification`, this is perhaps a better place for it (since it's a concern of the query rather than of the API per-se). Thoughts?

Reply:
> I could move the defaulting to the `HostSpecification`, this is perhaps a better place for it (since it's a concern of the query rather than of the API per-se). Thoughts?

ack - my misunderstanding. Thanks for the rundown. Ya, I think that would be a better place for it.